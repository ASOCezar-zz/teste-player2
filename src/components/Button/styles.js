import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    height: 43px;
    cursor: pointer;

    button {
      width: 100%;
      height: 100%;
      display: flex;
      justify-content: center;
      align-items: center;
      background-color: ${theme.colors.green};
      border: none;
      color: ${theme.colors.white};
      padding: 12px 24px;
      p {
        font-size: 14px;
        font-style: normal;
        font-weight: 700;
        line-height: 19px;
        text-align: left;
        padding: 0;
        margin: 0;
      }
    }
  `}
`;
