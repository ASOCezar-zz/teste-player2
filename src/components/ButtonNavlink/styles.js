import styled, { css, keyframes } from 'styled-components';

const openingOption = keyframes`
  0% {
    opacity: 0;
    transform: rotateX(-90deg);
  }
  100% {
    opacity: 1;
    transform: rotateX(0deg);
  }
`;

const openedOptions = (openList) =>
  openList
    ? css`
        width: 100%;
        perspective: 1000px;
        .drop-item {
          display: flex;
          flex-direction: column;
          align-items: flex-start;
          justify-content: center;
          padding: 12px 10px 12px 44px;
          height: 42px;
          opacity: 0;
          transform-origin: top center;
          backdrop-filter: brightness(80%);
          &:nth-child(1n) {
            animation: ${openingOption};
            animation-duration: 400ms;
            animation-delay: calc(150ms);
            animation-timing-function: ease-in-out;
            animation-fill-mode: forwards;
          }
          &:nth-child(2n) {
            animation: ${openingOption};
            animation-duration: 400ms;
            animation-delay: calc(150ms * 2);
            animation-timing-function: ease-in-out;
            animation-fill-mode: forwards;
          }

          a {
            font-size: 14px;
            font-style: normal;
            font-weight: 400;
            line-height: 18px;
            letter-spacing: 0em;
            text-align: left;
            cursor: pointer;
          }
        }
      `
    : css`
        .drop-item {
          display: none;
        }
      `;

export const Container = styled.div`
  ${({ theme, openList }) => css`
    .btn-container {
      width: 100%;
      display: flex;
      flex-direction: row;
      justify-content: space-between;
      align-items: center;
      transition: all 250ms linear;
      cursor: pointer;
      padding: 15px;
      user-select: none;

      .see-more {
        display: flex;
        flex-direction: row;
        gap: 15px;
        background-color: transparent;
        border: none;
        align-items: center;
        cursor: pointer;

        img {
          width: 24px;
          height: 24px;
        }

        span {
          color: ${theme.colors.white};
          font-size: 14px;
          font-weight: 400;
          line-height: 20px;
        }

        img {
          cursor: pointer;
        }
      }
    }

    .options {
      list-style: none;
      ${openedOptions(openList)}
    }
  `}
`;
