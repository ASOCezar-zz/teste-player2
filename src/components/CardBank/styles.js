import styled, { css, keyframes } from 'styled-components';

const showOption = keyframes`
  from {
    box-shadow: 0px 0px 0px 0px rgba(71, 79, 99, 0.15);
    bottom: 12px 
  }
  to {
    box-shadow: 0px 10px 15px 0px rgba(71, 79, 99, 0.15);
    bottom: 40px
  }
`;
const hideOption = keyframes`
  from {
    box-shadow: 0px 10px 15px 0px rgba(71, 79, 99, 0.15);
    bottom: 40px
  }
  to {
    box-shadow: 0px 0px 0px 0px rgba(71, 79, 99, 0.15);
    bottom: 12px
  }
`;

export const Container = styled.div`
  ${({ theme, openOptions }) => css`
    background-color: ${theme.colors.white};
    border-radius: 8px;
    box-shadow: 0px 4px 20px 0px rgba(71, 79, 99, 0.05);
    padding: 16px 20px 24px;
    max-width: 100%;
    max-height: 100%;
    box-sizing: border-box;

    transition: all 0.3s linear;

    &:hover {
      background-color: rgba(237, 239, 242, 1);
    }

    &:active {
      background-color: rgba(228, 231, 236, 1);
    }

    .head-card {
      display: flex;
      flex-direction: row;
      width: 100%;
      align-items: center;
      justify-content: space-between;

      .dispatching-card {
        display: flex;
        flex-direction: row;
        align-items: center;
        gap: 10px;
        cursor: pointer;
        label {
          color: ${theme.colors.blue};
          font-size: 12px;
          font-weight: 500;
        }
      }

      .profiles-card {
        display: flex;
        flex-direction: row;
        align-items: center;
        gap: 5px;
        cursor: pointer;
        span {
          color: ${theme.colors.secondaryFont};
          font-size: 14px;
          font-weight: 500;
        }
      }

      .options-card {
        .btn-options {
          position: relative;
          border: none;
          background-color: transparent;
          cursor: pointer;
          z-index: 3;

          .tooltip {
            position: absolute;
            display: flex;
            flex-direction: row;
            bottom: 40px;
            right: -30px;
            background-color: ${theme.colors.white};
            border-radius: 6px;
            padding: 5px 0;
            text-align: center;
            width: 172px;
            height: 52px;
            animation: ${openOptions ? showOption : hideOption} 250ms ease-in-out forwards;
            visibility: ${openOptions ? 'visible' : 'hidden'};

            align-items: center;

            transition: all 250ms ease-in-out;

            .tooltip-text {
              color: ${theme.colors.gray};
              display: flex;
              flex-direction: row;
              justify-content: space-around;
              align-items: center;
              width: 100%;

              padding: 8px 40px 8px 14px;

              &::after {
                content: '';
                position: absolute;
                top: 100%;
                left: 75%;
                margin-left: -5px;
                border-width: 5px;
                border-style: solid;
                border-color: ${theme.colors.white} transparent transparent transparent;
              }
            }
          }

          :active {
            background-color: ${theme.colors.darkGray};
            outline: none;
            user-select: none;
          }
        }
      }
    }

    .content-card {
      display: flex;
      flex-direction: row;
      gap: 12px;
      height: 100%;
      align-items: center;

      .txt-content-card {
        display: flex;
        flex-direction: column;
        gap: 5px;
        height: 40px;

        h4 {
          color: ${theme.colors.primaryFont};
        }
      }
    }
  `}
`;
