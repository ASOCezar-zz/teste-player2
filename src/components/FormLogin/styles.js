import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme, isEmailInvalid, isPasswordInvalid }) => css`
    width: 100%;
    form {
      display: flex;
      flex-direction: column;
      width: 100%;
      height: auto;
      gap: 12px;

      .form-group {
        display: flex;
        flex-direction: column;
        gap: 5px;


        .field-label {
          font-size: 12px;
          font-weight: 500;
          line-height: 16px;
          color: ${theme.colors.neutralDark};
          margin-left: 16px;
        }

        #email {
          border: ${!isEmailInvalid ? `1px solid ${theme.colors.darkGray};` : `1px solid ${theme.colors.error}`}
        }
        #password {
          border: ${!isPasswordInvalid ? `1px solid ${theme.colors.darkGray};` : `1px solid ${theme.colors.error}`}
        }

        .form-input {
          width: 100%;
          height: 44px;
          border-radius: 4px;
          padding: 12px 16px;
          outline: none;
          transition: box-shadow 250ms linear;

          ::placeholder {
            color: ${theme.colors.secondaryFont};
          }
          box-shadow: 0px 0px 0px 0px rgba(71, 79, 99, 0.15);
          
          &:focus {
              box-shadow: 0px 10px 15px 0px rgba(71, 79, 99, 0.15);
            }
        }

        .form-error {
          margin-left: 16px;
          display: flex;
          flex-direction: row;
          gap: 5.75px;
          align-items: center;

            p {
              color: ${theme.colors.error};
              font-size: 11px;
              weight: 500;
            }
          }
        }
        
        .btn-login {
          width: 100%;
          height: 43px;
          border: none;

          cursor: pointer;
          background-color: ${theme.colors.green};
          color: ${theme.colors.white};
          font-size: 14px;
          font-weight: 700;
          line-height: 19px;

          &:hover {
            background-color: rgba(14, 195, 163, 0.7);
          }

          &:active {
            background-color: rgba(87, 180, 163, 1);
          }
        }
      }
    }
  `}
`;
