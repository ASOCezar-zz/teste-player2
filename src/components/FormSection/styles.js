import styled, { css } from 'styled-components';

export const Container = styled.section`
  ${({ theme }) => css`
    min-width: 50%;
    padding: 89px 24px 48px;
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 48px;

    ${theme.media.desktop} {
      padding: 89px 10%;
    }

    .logo {
      width: 107.19px;
      height: 84px;

      img {
        width: 100%;
        height: 100%;
      }
    }

    .login-title {
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      text-align: center;

      h2 {
        color: ${theme.colors.primaryFont};
        font-size: 30px;
        line-height: 32px;
      }
    }

    .register {
      display: inline-block;

      .register-text {
        font-size: 14px;
        font-weight: 400;
        color: ${theme.colors.gray};
        line-height: 19px;

        .registration-link {
          text-decoration: none;
          color: ${theme.colors.green};
          font-weight: 700;
          line-height: 20px;
        }
      }
    }

    footer {
      position: absolute;
      bottom: 20px;
      font-size: 14px;
      font-style: normal;
      font-weight: 500;
      line-height: 18px;
      letter-spacing: 0em;
      text-align: center;
      color: #c6cad6;
    }
  `}
`;
