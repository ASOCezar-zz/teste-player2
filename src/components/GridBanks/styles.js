import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    max-width: 100%;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
    grid-auto-rows: 160px;
    gap: 12px;
  `}
`;
