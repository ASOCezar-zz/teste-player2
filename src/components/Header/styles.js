import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    position: sticky;
    inset: 0 0 0 24px;
    width: 100%;
    display: flex;
    flex-direction: row;
    header {
      display: flex;
      flex-direction: row;
      width: 100%;
      height: 44px;
      gap: 24px;
      top: 0;
      left: 0;
      background-color: ${theme.colors.background};
      align-items: center;
      justify-content: space-between;
      z-index: 2;
      flex-wrap: nowrap;
      border-bottom: 1px solid ${theme.colors.secondaryFont};

      .funcionalities {
        display: none;
        visibility: hidden;
      }

      ${theme.media.desktop} {
        padding: 18.5px;
        height: 80px;
        .funcionalities {
          display: flex;
          visibility: visible;
          gap: 30px;

          .btn-profile {
            padding: 0;
            margin: 0;
            display: flex;
            flex-direction: row;
            background-color: transparent;
            border: none;
            gap: 11px;
            align-items: center;
            justify-content: center;

            p {
              font-size: 14px;
              font-weight: 400;
              line-height: 20px;
              color: ${theme.colors.gray};

              strong {
                font-weight: 700;
                line-height: 19px;
                color: ${theme.colors.green};
              }
            }
          }
        }
      }

      .top-page {
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: flex-start;

        img {
          cursor: pointer;
        }

        h3 {
          color: ${theme.colors.secondaryFont};
          font-size: 20px;
          font-style: normal;
          font-weight: 500;
          line-height: 32px;
          letter-spacing: 0em;
          white-space: nowrap;
        }
      }

      .page-title {
        height: 100%;
        display: flex;
        align-items: center;
        overflow: hidden;
      }
    }
  `}
`;
