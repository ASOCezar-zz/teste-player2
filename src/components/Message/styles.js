import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    flex-direction: column;
    gap: 4px;

    .message-title {
      font-size: 12px;
      line-height: 16px;
      color: ${theme.colors.neutralDark};
    }

    .message-text {
      color: ${theme.colors.primaryFont};
      font-weight: 400;
      line-height: 20px;
      font-size: 14px;
    }

    .message-attachment {
      display: flex;
      flex-direction: row;
      gap: 6.25px;

      p {
        font-size: 12px;
        font-weight: 400;
        line-height: 16px;
        color: ${theme.colors.gray};
      }
    }
  `}
`;
