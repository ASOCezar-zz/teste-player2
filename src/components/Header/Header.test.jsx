import { screen } from '@testing-library/react';
import { Header } from '.';
import { renderTheme } from '../../styles/renderTheme';

describe('<Header />', () => {
  it('Should render a header with an alert for notifications', () => {
    renderTheme(<Header pageTitle={<h1>Teste</h1>} />);
    expect(screen.getByRole('heading', { name: 'Teste' })).toBeInTheDocument();
    expect(screen.getByAltText('icone sem notificações')).toBeInTheDocument();
    expect(screen.queryByAltText('icone com notificações')).not.toBeInTheDocument();
  });
  it('Should match with a snapshot', () => {
    const { container } = renderTheme(<Header pageTitle="Teste" />);
    expect(container).toMatchSnapshot();
  });
});
