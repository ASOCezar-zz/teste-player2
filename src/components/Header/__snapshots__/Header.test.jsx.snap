// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should match with a snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 43px;
}

.c1 button {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #0EC3A3;
  border: none;
  color: #FFFFFF;
  padding: 12px 24px;
  cursor: pointer;
}

.c1 button:hover {
  background-color: rgba(14,195,163,0.7);
}

.c1 button:active {
  background-color: rgba(87,180,163,1);
}

.c1 button p {
  font-size: 14px;
  font-style: normal;
  font-weight: 700;
  line-height: 19px;
  text-align: left;
  padding: 0;
  margin: 0;
}

.c0 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  left: 0;
  height: 44px;
  margin-left: 44px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 24px;
  background-color: #f7f7f8;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  z-index: 2;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  border-bottom: 1px solid #b3b9c9;
}

.c0 .funcionalities {
  display: none;
  visibility: hidden;
}

.c0 .top-page {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .top-page .title-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 .top-page .title-content img {
  cursor: pointer;
}

.c0 .top-page .title-content h3 {
  color: #b3b9c9;
  font-size: 20px;
  font-style: normal;
  font-weight: 500;
  line-height: 32px;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  white-space: nowrap;
}

.c0 .page-title {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
}

@media (min-width:1024px) {
  .c0 {
    margin-left: 0;
    height: 80px;
  }

  .c0 .funcionalities {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    visibility: visible;
    gap: 30px;
  }

  .c0 .funcionalities .notification-button {
    cursor: pointer;
  }

  .c0 .funcionalities .btn-profile {
    padding: 0;
    margin: 0;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    background-color: transparent;
    border: none;
    gap: 11px;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    cursor: pointer;
  }

  .c0 .funcionalities .btn-profile p {
    font-size: 14px;
    font-weight: 400;
    line-height: 20px;
    color: #6e778d;
  }

  .c0 .funcionalities .btn-profile p strong {
    font-weight: 700;
    line-height: 19px;
    color: #0EC3A3;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="page-title"
    >
      <div
        class="top-page"
      >
        <div
          class="title-content"
        >
          Teste
        </div>
      </div>
    </div>
    <div
      class="funcionalities"
    >
      <div
        class="c1"
      >
        <button>
          <img
            src=""
          />
          <p>
            + Criar Banco
          </p>
        </button>
      </div>
      <img
        alt="icone sem notificações"
        class="notification-button"
        src="withoutNotification-icon.svg"
      />
      <button
        class="btn-profile"
      >
        <p>
          Olá, 
          <strong>
            Beleza Rara
          </strong>
        </p>
        <img
          src="arrowDown-icon.svg"
        />
      </button>
    </div>
  </header>
</div>
`;
